cranelift/codegen/meta/src/shared/settings.rs

```diff
@@ -3,39 +3,6 @@ use crate::cdsl::settings::{SettingGroup, SettingGroupBuilder};
 pub(crate) fn define() -> SettingGroup {
     let mut settings = SettingGroupBuilder::new("shared");
 
-    settings.add_enum(
-        "regalloc",
-        "Register allocator to use with the MachInst backend.",
-        r#"
-            This selects the register allocator as an option among those offered by the `regalloc.rs`
-            crate. Please report register allocation bugs to the maintainers of this crate whenever
-            possible.
-
-            Note: this only applies to target that use the MachInst backend. As of 2020-04-17, this
-            means the x86_64 backend doesn't use this yet.
-
-            Possible values:
-
-            - `backtracking` is a greedy, backtracking register allocator as implemented in
-            Spidermonkey's optimizing tier IonMonkey. It may take more time to allocate registers, but
-            it should generate better code in general, resulting in better throughput of generated
-            code.
-            - `backtracking_checked` is the backtracking allocator with additional self checks that may
-            take some time to run, and thus these checks are disabled by default.
-            - `experimental_linear_scan` is an experimental linear scan allocator. It may take less
-            time to allocate registers, but generated code's quality may be inferior. As of
-            2020-04-17, it is still experimental and it should not be used in production settings.
-            - `experimental_linear_scan_checked` is the linear scan allocator with additional self
-            checks that may take some time to run, and thus these checks are disabled by default.
-        "#,
-        vec![
-            "backtracking",
-            "backtracking_checked",
-            "experimental_linear_scan",
-            "experimental_linear_scan_checked",
-        ],
```

Comment:
How to enable the regalloc checker without this option? And what about potential other regallocs?

Reply:
> what about potential other regallocs

Right now regalloc2 doesn't have any other algorithms, but we could certainly add this option back if it gains any. (And, for maximal clarity here, retaining the existing options is unfortunately not really possible because regalloc.rs is so different in certain ways from regalloc2 -- we can't make it a runtime option, we have to switch over all at once.)

> How to enable the regalloc checker

I actually need to do a bit more work if we want the checker enabled *in Cranelift* (as opposed to in the regalloc2 fuzzing harness) -- it doesn't support pinned vregs currently, which the Cranelift glue uses to model RealRegs. I could do that, I just haven't had time yet; not sure if it should block an initial merge or not (open to opinions on this...).

Reply:
Filed bytecodealliance/regalloc2#42 for this.